hapi-proto/src/main/java/com/hedera/services/legacy/proto/utils/CommonUtils.java

```diff
@@ -234,48 +235,79 @@ public static String toReadableString(Transaction grpcTransaction) throws Invali
-    String rv = null;
-    TransactionBody body;
-    if (grpcTransaction.hasBody()) {
-      body = grpcTransaction.getBody();
-    } else {
-      body = TransactionBody.parseFrom(grpcTransaction.getBodyBytes());
-    }
-    rv = "txID=" + TextFormat.shortDebugString(body.getTransactionID()) + "; memo=" + body
-            .getMemo();
+    TransactionBody body = extractTransactionBody(grpcTransaction);
+    String rv = "txID=" + TextFormat.shortDebugString(body.getTransactionID());
     return rv;
   }
 
-  /**
-   * Generates a short human readable string for grpc transaction ID.
-   *
-   * @return generated readable string
-   */
-  public static String toReadableStringShortTxnID(
-          Transaction grpcTransaction) throws InvalidProtocolBufferException {
-    String rv = null;
-    TransactionBody body;
-    if (grpcTransaction.hasBody()) {
-      body = grpcTransaction.getBody();
-    } else {
-      body = TransactionBody.parseFrom(grpcTransaction.getBodyBytes());
+  public static ByteString extractTransactionBodyByteString(TransactionOrBuilder transaction)
+      throws InvalidProtocolBufferException {
+    ByteString signedTransactionBytes = transaction.getSignedTransactionBytes();
+    if (!signedTransactionBytes.isEmpty()) {
+      return SignedTransaction.parseFrom(signedTransactionBytes).getBodyBytes();
     }
-    rv = "txID=" + TextFormat.shortDebugString(body.getTransactionID());
-    return rv;
+
+    return transaction.getBodyBytes();
   }
 
+  public static byte[] extractTransactionBodyBytes(TransactionOrBuilder transaction)
+          throws InvalidProtocolBufferException {
+    return extractTransactionBodyByteString(transaction).toByteArray();
+  }
 
-  public static TransactionBody extractTransactionBody(Transaction transaction)
-      throws InvalidProtocolBufferException {
-    TransactionBody bodyToReturn;
-    if (transaction.hasBody()) {
-      bodyToReturn = transaction.getBody();
-    } else {
-      bodyToReturn = TransactionBody.parseFrom(transaction.getBodyBytes());
+  public static TransactionBody extractTransactionBody(TransactionOrBuilder transaction)
+          throws InvalidProtocolBufferException {
+    return TransactionBody.parseFrom(extractTransactionBodyByteString(transaction));
+  }
+
+  public static SignatureMap extractSignatureMap(TransactionOrBuilder transaction)
+          throws InvalidProtocolBufferException {
+    ByteString signedTransactionBytes = transaction.getSignedTransactionBytes();
+    if (!signedTransactionBytes.isEmpty()) {
+      return SignedTransaction.parseFrom(signedTransactionBytes).getSigMap();
     }
-    return bodyToReturn;
+
+    return transaction.getSigMap();
+  }
+
+  public static SignatureMap extractSignatureMapOrUseDefault(TransactionOrBuilder transaction) {
+    try {
+      return extractSignatureMap(transaction);
+    } catch (InvalidProtocolBufferException ignoreToReturnDefault) { }
+    return SignatureMap.getDefaultInstance();
```

Comment:
I am not sure if I am following the logic here. I assume SignatureMap's default instance is still a legitimate instance. If the parsing of the transaction has throw an exception already, should we stop there already instead of provide the default instance of SignatureMap to continue. In other words, do we have a use case that needs this logic here?